### PR TITLE
Fix: Crossbrowser Chrome, Firefox and Edge tests

### PR DESCRIPTION
**Pull request by Rob-Doherty**
# Description

This change includes everything needed to get the crossbrowser tests running stably for Chrome, Firefox and Edge via Saucelabs:
- improved waiting 1: to make use of the `waitForTimeout` config rather than hardcoded timeouts
- improved waiting 2: wait for certain conditions rather than just check for them immediately
- refactor crossbrowser config to W3C standard for `capabilities`
- skipped a SessionHelper.js check if running crossbrowser tests as this is already covered by the build pipeline e2e tests
- upgrade WebdriverIO to v6

Fixes # 
https://tools.hmcts.net/jira/browse/TA-162
https://tools.hmcts.net/jira/browse/TA-164

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

This has been tested successfully many times locally pointing to AAT via Saucelabs.

Successfully ran in nightly build (after some retries): https://build.platform.hmcts.net/job/HMCTS_Nightly_DIV/job/div-petitioner-frontend/job/PR-774/3/

chrome, firefox and edge success nightly

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: test/end-to-end/helpers/SessionHelper.js [hunk 1]
```diff
@@ -68,20 +68,25 @@ class SessionHelper extends codecept_helper {
...
+      const authTokenCookie = await helper.grabCookie('__auth-token');
+      const session = await this.getTheSession(connectSidCookie, authTokenCookie);
+      session.featureToggles.ft_welsh = true;
+      session.petitionerPcqId = 'is_in_session';
+      session.featureToggles.ft_awaiting_amend = true;
+
+      let expectedSession = this.updateExpectedSessionWithActualSession(basicDivorceSessionData, session);
+
+      return assert.deepEqual(session, expectedSession);
+    }
```
Comment (Rob-Doherty) [inline test/end-to-end/helpers/SessionHelper.js:89]: This changes isn't needed, there's a feature toggle to turn it off via the env variable `IGNORE_SESSION_VALIDATION` - need to revert.

File: test/end-to-end/pages/address/postcode.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ function enterAddressUsingPostcode(stepUrl, testAddressIndex) {
   if (!testAddressIndex) {
     testAddressIndex = '0';
   }
-  I.waitInUrl(stepUrl, 5);
+  I.waitInUrl(stepUrl);
```
Comment (mounikahmcts) [inline test/end-to-end/pages/address/postcode.js:7]: why did we remove delay here? Delay 5 sec could have added purposefully as sometimes we have performance isuues

File: test/end-to-end/pages/financial/advice.js [hunk 3]
```diff
@@ -3,8 +3,8 @@ const pagePath = '/about-divorce/financial/advice';
 function enterFinancialAdvice() {
 
   const I = this;
-  I.waitInUrl(pagePath, 5);
+  I.waitInUrl(pagePath);
```
Comment (mounikahmcts) [inline test/end-to-end/pages/financial/advice.js:6]: same here

File: test/end-to-end/pages/address/postcode.js (hunk 2 above)
Comment (Rob-Doherty) [reply test/end-to-end/pages/address/postcode.js:7]: The delay here was hardcoded to `5`, but in the config it's already set to 5 seconds
```
e2e:
  ...
  waitForTimeoutValue: 5000
  ...
```
https://github.com/hmcts/div-petitioner-frontend/blob/master/config/default.yaml#L159

However this is  a problem for the veeery slow crossbrowser tests that run remotely on saucelabs, as they often exceed these hardcoded timeouts. So instead, by not supplying a timeout the step now falls back to using the runtime `waitForTimeout` which can be set as short for headless e2e tests (e.g. 5 seconds) or longer for overnight tests (e.g. 20 seconds) via either the default config or in the Codecept config.

File: test/end-to-end/pages/financial/advice.js (hunk 3 above)
Comment (Rob-Doherty) [reply test/end-to-end/pages/financial/advice.js:6]: see above

File: test/end-to-end/pages/address/postcode.js (hunk 2 above)
Comment (Rob-Doherty) [reply test/end-to-end/pages/address/postcode.js:7]: The CodeceptJS docs can be a little tricky to navigate sometimes, but [here's a link](https://codecept.io/helpers/Puppeteer/#configuration) to the Pupeteer helper docs where you can see reference to the global `waitForTimeout`

Comment (mounikahmcts) [reply test/end-to-end/pages/address/postcode.js:7]: 👍

**Review by mounikahmcts — APPROVED** [review]

**Comment by Rob-Doherty** [comment]
Closing this PR as it was easier to cut a new branch and PR after rebasing off the Welsh refactor: https://github.com/hmcts/div-petitioner-frontend/pull/779

**Pull request closed without merging**